### PR TITLE
CASMHMS-5870 HBTD Update to separate ETCD chart

### PR DESCRIPTION
**Pull request by nieuwsma**
## Summary and Scope

convert etcd operator

## Issues and Related PRs

* Resolves [CASMHMS-5870](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-5870)

## Testing

See comment below for log

Tested on:

  * Mug
  * 
Test description:

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? N
- Were continuous integration tests run? If not, why? N
- Was upgrade tested? If not, why? NO, NA yet
- Was downgrade tested? If not, why? NO, NA yet

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

no 
## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

**Comment by nieuwsma** [comment]
`
ncn-m001:~/anieuwsma # helm uninstall -n services cray-hms-hbtd
release "cray-hms-hbtd" uninstalled
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-9d685dbf7-b2sv9                                         2/2     Terminating             0          89d
cray-hbtd-9d685dbf7-grjgn                                         2/2     Terminating             0          89d
cray-hbtd-9d685dbf7-qxpkp                                         2/2     Terminating             0          87d
cray-hbtd-etcd-ll7dhzsrh4                                         0/1     Terminating             1          87d
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
ncn-m001:~/anieuwsma # helm upgrade -i -n services cray-hms-hbtd cray-hms-hbtd-2.1.4-20230110165306+067b6ff.tgz
Release "cray-hms-hbtd" does not exist. Installing it now.
NAME: cray-hms-hbtd
LAST DEPLOYED: Wed Jan 11 14:15:09 2023
NAMESPACE: services
STATUS: deployed
REVISION: 1
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-wait-for-etcd-1-42zkp                                   1/1     Running                 0          4s
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-5f9cc764cc-9c978                                        0/2     Init:0/2                0          11s
cray-hbtd-5f9cc764cc-ljwz7                                        0/2     Init:0/2                0          4s
cray-hbtd-5f9cc764cc-rvtfg                                        0/2     Init:0/2                0          6s
cray-hbtd-etcd-0                                                  0/2     Init:0/1                0          10s
cray-hbtd-etcd-1                                                  0/2     Init:0/1                0          5s
cray-hbtd-wait-for-etcd-1-42zkp                                   1/1     Running                 0          17s
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-5f9cc764cc-9c978                                        0/2     Init:0/2                0          46s
cray-hbtd-5f9cc764cc-ljwz7                                        0/2     Init:0/2                0          39s
cray-hbtd-5f9cc764cc-rvtfg                                        0/2     Init:0/2                0          41s
cray-hbtd-etcd-0                                                  1/2     Running                 0          45s
cray-hbtd-etcd-1                                                  1/2     Running                 0          40s
cray-hbtd-etcd-2                                                  1/2     Running                 0          31s
cray-hbtd-wait-for-etcd-1-42zkp                                   1/1     Running                 0          52s
ncn-m001:~/anieuwsma # watch "kubectl -n services get pods | grep hbtd"
ncn-m001:~/anieuwsma # kubectl -n services logs cray-hbtd-5f9cc764cc-9c978
2023/01/11 14:17:25 Cray Node Heartbeat Monitor/Tracker started.
2023/01/11 14:17:25 Service name: 'cray-hbtd-5f9cc764cc-9c978'
2023/01/11 14:17:25 Waiting for HSM to be responsive...
2023/01/11 14:17:26 HSM is responsive.
INFO: Setting KV URL from ETCD_HOST and ETCD_PORT (http://cray-hbtd-etcd:2379)
2023/01/11 14:17:28 ETCD connection succeeded.
2023/01/11 14:17:29 K/V health check succeeded.
2023/01/11 14:17:29 INFO: No life keys found, HB key cleanup set to: 0.
2023/01/11 14:17:29 Listening on port 28500
time="2023-01-11T14:17:29Z" level=info msg="Message bus connect, kafka, confluent interface."
2023/01/11 14:17:29 INFO: Starting up HTTP server.
time="2023-01-11T14:17:29Z" level=info msg="Connected to Confluent Kafka server as Writer."
2023/01/11 14:17:29 Connected to Telemetry Bus.
2023/01/11 14:17:29 Life key 'hbtd_lifekey-1740766521' created.
%3|1673446650.096|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1021ms in state CONNECT)
%3|1673446652.080|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1005ms in state CONNECT, 1 identical error(s) suppressed)
2023/01/11 14:17:56 INFO: Heartbeat started for 'x3001c0s35b0n0'
2023/01/11 14:17:56 INFO: Heartbeat started for 'x3000c0s11b0n0'
2023/01/11 14:17:57 INFO: Heartbeat started for 'x3000c0s5b0n0'
2023/01/11 14:17:59 Number of components heartbeating: 4
2023/01/11 14:17:59 INFO: Heartbeat started for 'x3000c0s26b0n0'
2023/01/11 14:17:59 INFO: Heartbeat started for 'x3000c0s17b0n0'
2023/01/11 14:18:02 INFO: Heartbeat started for 'x3000c0s15b0n0'
2023/01/11 14:18:02 INFO: Heartbeat started for 'x3001c0s33b0n0'
%3|1673446683.088|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1012ms in state CONNECT, 16 identical error(s) suppressed)
2023/01/11 14:18:03 INFO: Heartbeat started for 'x3000c0s1b0n0'
2023/01/11 14:18:04 Number of components heartbeating: 13
2023/01/11 14:18:09 Number of components heartbeating: 16
%3|1673446713.104|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1027ms in state CONNECT, 15 identical error(s) suppressed)
ncn-m001:~/anieuwsma #
`

**Comment by nieuwsma** [comment]
```
ncn-m001:~/anieuwsma # helm uninstall -n services cray-hms-hbtd
release "cray-hms-hbtd" uninstalled
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-9d685dbf7-b2sv9                                         2/2     Terminating             0          89d
cray-hbtd-9d685dbf7-grjgn                                         2/2     Terminating             0          89d
cray-hbtd-9d685dbf7-qxpkp                                         2/2     Terminating             0          87d
cray-hbtd-etcd-ll7dhzsrh4                                         0/1     Terminating             1          87d
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
ncn-m001:~/anieuwsma # helm upgrade -i -n services cray-hms-hbtd cray-hms-hbtd-2.1.4-20230110165306+067b6ff.tgz
Release "cray-hms-hbtd" does not exist. Installing it now.
NAME: cray-hms-hbtd
LAST DEPLOYED: Wed Jan 11 14:15:09 2023
NAMESPACE: services
STATUS: deployed
REVISION: 1
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-wait-for-etcd-1-42zkp                                   1/1     Running                 0          4s
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-5f9cc764cc-9c978                                        0/2     Init:0/2                0          11s
cray-hbtd-5f9cc764cc-ljwz7                                        0/2     Init:0/2                0          4s
cray-hbtd-5f9cc764cc-rvtfg                                        0/2     Init:0/2                0          6s
cray-hbtd-etcd-0                                                  0/2     Init:0/1                0          10s
cray-hbtd-etcd-1                                                  0/2     Init:0/1                0          5s
cray-hbtd-wait-for-etcd-1-42zkp                                   1/1     Running                 0          17s
ncn-m001:~/anieuwsma # kubectl -n services get pods | grep hbtd
cray-hbtd-5f9cc764cc-9c978                                        0/2     Init:0/2                0          46s
cray-hbtd-5f9cc764cc-ljwz7                                        0/2     Init:0/2                0          39s
cray-hbtd-5f9cc764cc-rvtfg                                        0/2     Init:0/2                0          41s
cray-hbtd-etcd-0                                                  1/2     Running                 0          45s
cray-hbtd-etcd-1                                                  1/2     Running                 0          40s
cray-hbtd-etcd-2                                                  1/2     Running                 0          31s
cray-hbtd-wait-for-etcd-1-42zkp                                   1/1     Running                 0          52s
ncn-m001:~/anieuwsma # watch "kubectl -n services get pods | grep hbtd"
ncn-m001:~/anieuwsma # kubectl -n services logs cray-hbtd-5f9cc764cc-9c978
2023/01/11 14:17:25 Cray Node Heartbeat Monitor/Tracker started.
2023/01/11 14:17:25 Service name: 'cray-hbtd-5f9cc764cc-9c978'
2023/01/11 14:17:25 Waiting for HSM to be responsive...
2023/01/11 14:17:26 HSM is responsive.
INFO: Setting KV URL from ETCD_HOST and ETCD_PORT (http://cray-hbtd-etcd:2379)
2023/01/11 14:17:28 ETCD connection succeeded.
2023/01/11 14:17:29 K/V health check succeeded.
2023/01/11 14:17:29 INFO: No life keys found, HB key cleanup set to: 0.
2023/01/11 14:17:29 Listening on port 28500
time="2023-01-11T14:17:29Z" level=info msg="Message bus connect, kafka, confluent interface."
2023/01/11 14:17:29 INFO: Starting up HTTP server.
time="2023-01-11T14:17:29Z" level=info msg="Connected to Confluent Kafka server as Writer."
2023/01/11 14:17:29 Connected to Telemetry Bus.
2023/01/11 14:17:29 Life key 'hbtd_lifekey-1740766521' created.
%3|1673446650.096|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1021ms in state CONNECT)
%3|1673446652.080|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1005ms in state CONNECT, 1 identical error(s) suppressed)
2023/01/11 14:17:56 INFO: Heartbeat started for 'x3001c0s35b0n0'
2023/01/11 14:17:56 INFO: Heartbeat started for 'x3000c0s11b0n0'
2023/01/11 14:17:57 INFO: Heartbeat started for 'x3000c0s5b0n0'
2023/01/11 14:17:59 Number of components heartbeating: 4
2023/01/11 14:17:59 INFO: Heartbeat started for 'x3000c0s26b0n0'
2023/01/11 14:17:59 INFO: Heartbeat started for 'x3000c0s17b0n0'
2023/01/11 14:18:02 INFO: Heartbeat started for 'x3000c0s15b0n0'
2023/01/11 14:18:02 INFO: Heartbeat started for 'x3001c0s33b0n0'
%3|1673446683.088|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1012ms in state CONNECT, 16 identical error(s) suppressed)
2023/01/11 14:18:03 INFO: Heartbeat started for 'x3000c0s1b0n0'
2023/01/11 14:18:04 Number of components heartbeating: 13
2023/01/11 14:18:09 Number of components heartbeating: 16
%3|1673446713.104|FAIL|rdkafka#producer-1| [thrd:cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap]: cluster-kafka-bootstrap.sma.svc.cluster.local:9092/bootstrap: Connect to ipv4#10.27.190.248:9092 failed: Connection refused (after 1027ms in state CONNECT, 15 identical error(s) suppressed)
ncn-m001:~/anieuwsma #
ncn-m001:~/anieuwsma # sat status
+----------------+-----------+------+----------+-----------+---------+---------+------+-------+-------------+---------+----------+-----------------------------+----------------------+-------------+
| xname          | Aliases   | Type | NID      | State     | Flag    | Enabled | Arch | Class | Role        | SubRole | Net Type | Desired Config              | Configuration Status | Error Count |
+----------------+-----------+------+----------+-----------+---------+---------+------+-------+-------------+---------+----------+-----------------------------+----------------------+-------------+
| x3000c0s1b0n0  | ncn-m001  | Node | 100001   | Populated | OK      | True    | X86  | River | Management  | Master  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s3b0n0  | ncn-m002  | Node | 100002   | Ready     | OK      | True    | X86  | River | Management  | Master  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s5b0n0  | ncn-m003  | Node | 100003   | Ready     | OK      | True    | X86  | River | Management  | Master  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s7b0n0  | ncn-w001  | Node | 100004   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s9b0n0  | ncn-w002  | Node | 100005   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s11b0n0 | ncn-w003  | Node | 100006   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s13b0n0 | ncn-s001  | Node | 100007   | Ready     | OK      | True    | X86  | River | Management  | Storage | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s15b0n0 | ncn-s002  | Node | 100008   | Ready     | OK      | True    | X86  | River | Management  | Storage | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s17b0n0 | ncn-s003  | Node | 100009   | Ready     | OK      | True    | X86  | River | Management  | Storage | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s19b1n0 | nid000001 | Node | 1        | Off       | OK      | True    | X86  | River | Compute     | None    | Sling    |                             | unconfigured         | 0           |
| x3000c0s19b2n0 | nid000002 | Node | 2        | On        | OK      | True    | X86  | River | Compute     | None    | Sling    |                             | unconfigured         | 0           |
| x3000c0s19b3n0 | nid000003 | Node | 3        | On        | OK      | True    | X86  | River | Compute     | None    | Sling    |                             | unconfigured         | 0           |
| x3000c0s19b4n0 | nid000004 | Node | 4        | On        | OK      | True    | X86  | River | Compute     | None    | Sling    |                             | unconfigured         | 0           |
| x3000c0s26b0n0 | ncn-s004  | Node | 100010   | Ready     | OK      | True    | X86  | River | Management  | Storage | Sling    | ncn-personalization         | pending              | 0           |
| x3000c0s28b0n0 | ncn-s005  | Node | 100011   | On        | OK      | True    | X86  | River | Management  | Storage | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s32b0n0 | uan01     | Node | 49169408 | Off       | OK      | True    | X86  | River | Application | UAN     | Sling    |                             | configured           | 0           |
| x3000c0s34b0n0 | ncn-w004  | Node | 100012   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3000c0s36b0n0 | ncn-w005  | Node | 100013   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | pending              | 0           |
| x3000c0s38b0n0 | ncn-w006  | Node | 100014   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3001c0s31b0n0 | ncn-w007  | Node | 100015   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3001c0s33b0n0 | ncn-w008  | Node | 100016   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3001c0s35b0n0 | ncn-w009  | Node | 100017   | Ready     | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization-abarry1 | failed               | 3           |
| x3001c0s37b0n0 | ncn-w010  | Node | 100018   | On        | Warning | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 3           |
| x3001c0s39b0n0 | ncn-w011  | Node | 100019   | On        | OK      | True    | X86  | River | Management  | Worker  | Sling    | ncn-personalization         | failed               | 4           |
+----------------+-----------+------+----------+-----------+---------+---------+------+-------+-------------+---------+----------+-----------------------------+----------------------+-------------+
ncn-m001:~/anieuwsma #
```

File: cray-hms-hbtd.compatibility.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ chartVersionToCSMVersion:
   # Chart version: CSM version
   ">=2.0.0": "~1.2.0" # Chart Version: 2.0.0 <= x.y.z, CSM Version: 1.2.0 <= x.y.z < 1.3.0
   ">=2.1.0": "~1.3.0"
+  ">=3.0.0": "#~1.6.0"
```
Comment (shunr-hpe) [inline cray-hms-hbtd.compatibility.yaml]: Is the `#` a typo

File: charts/v3.0/cray-hms-hbtd/values.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# Please refer to https://stash.us.cray.com/projects/CLOUD/repos/cray-charts/browse/stable/cray-service/values.yaml?at=refs%2Fheads%2Fmaster
```
Comment (shunr-hpe) [inline charts/v3.0/cray-hms-hbtd/values.yaml]: Is this now the url to go to?
https://github.com/Cray-HPE/base-charts/blob/master/kubernetes/cray-service/values.yaml

**Review by shunr-hpe — APPROVED** [review]

**Review by rsjostrand-hpe — APPROVED** [review]

**Comment by mjendrysik-hpe** [comment]
Completed what I could for HBTD on asthon. There are several things not working out there so we can't see heartbeats flow into HBTD.

```
ncn-m002:~/mjendrysik # kubectl get pods -n services | egrep hbtd
cray-hbtd-5d8f8789bb-4shj7                                        2/2     Running           0          3m15s
cray-hbtd-bitnami-etcd-0                                          2/2     Running           0          2m57s
cray-hbtd-bitnami-etcd-1                                          0/2     PodInitializing   0          12s
cray-hbtd-bitnami-etcd-2                                          2/2     Running           0          87s
cray-hbtd-etcd-post-install-b62l8                                 0/1     Completed         0          3m16s
cray-hbtd-pre-upgrade-etcd-backup-jplkb                           0/1     Completed         0          3m22s
cray-hbtd-wait-for-etcd-2-87qwj                                   0/1     Completed         0          129m
cray-hbtd-wait-for-etcd-5-tg4tg                                   0/1     Completed         0          3m17s
kube-etcd-defrag-cray-hbtd-etcd-28014720-2pfvf                    0/2     Completed         0          159m
```

**Pull request merged**